kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeBlock.kt

```diff
@@ -361,9 +363,23 @@ public class CodeBlock private constructor(
       else -> throw IllegalArgumentException("expected name but was $o")
     }
 
-    private fun argToLiteral(o: Any?) = o
+    private fun argToLiteral(o: Any?) = if (o is Number) formatNumericValue(o) else o
 
-    private fun argToString(o: Any?) = o?.toString()
+    private fun argToString(o: Any?) = if (o is Number) formatNumericValue(o) else o?.toString()
+
+    private fun formatNumericValue(o: Number): Any? {
+      val format = DecimalFormatSymbols().apply {
+        decimalSeparator = '.'
+        groupingSeparator = '_'
+      }
+
+      val formattedValue = DecimalFormat("###,###.####", format).format(o)
```

Comment:
Not really familiar with `DecimalFormat`'s pattern rules, I see in the tests that this pattern handles numbers with more than 6 digits, will it also properly handle numbers with more than 4 decimal digits? Wanna add a test case for this?